packages/app-store/salesforce/lib/CrmService.ts

```diff
@@ -711,4 +717,106 @@ export default class SalesforceCRMService implements CRM {
 
     return this.getDominantAccountId(response.records as { AccountId: string }[]);
   }
+
+  private setFallbackToContact(boolean: boolean) {
+    this.fallbackToContact = boolean;
+  }
+
+  private getFallbackToContact() {
+    return this.fallbackToContact;
+  }
+
+  private async writeToPersonRecord(
+    contactId: string,
+    startTime: string,
+    organizerEmail: string,
+    bookingUid?: string
+  ) {
+    const conn = await this.conn;
+    const appOptions = this.getAppOptions();
+    const { createEventOn, onBookingWriteToRecordFields } = appOptions;
+
+    // Figure out what we're writing to
+    let personRecordType = appOptions.createEventOn;
+
+    if (
+      createEventOn === SalesforceRecordEnum.LEAD &&
+      appOptions.createEventOnLeadCheckForContact &&
```

Comment:
This option allows us to write to either a lead or a contact. So we need to determine what record to query.